### PR TITLE
Update pickle.rst

### PR DESCRIPTION
**Pull request by geryogam**
Fix some typos in the documentation of the `pickle` standard library.

### CODE REVIEW

File: Doc/library/pickle.rst [hunk 1]
```diff
@@ -494,20 +494,18 @@ What can be pickled and unpickled?
...
-* functions defined at the top level of a module (using :keyword:`def`, not
-  :keyword:`lambda`)
+* tuples, lists, sets, and dictionaries containing only picklable objects;
 
-* built-in functions defined at the top level of a module
+* functions (built-in and user-defined) defined at the top level of a module
+  (using :keyword:`def`, not :keyword:`lambda`);
 
-* classes that are defined at the top level of a module
+* classes defined at the top level of a module;
```
Comment (JelleZijlstra) [inline Doc/library/pickle.rst:508]: This isn't actually true, classes nested in other classes work fine. Opened https://bugs.python.org/issue47206 to fix this.

Comment (geryogam) [reply Doc/library/pickle.rst:508]: Good catch! Should I fix it in this PR?

Comment (maggyero) [reply Doc/library/pickle.rst:508]: Good catch! Should I fix it in this PR?

Comment (JelleZijlstra) [reply Doc/library/pickle.rst:508]: Let's do it separately, so we can keep this to non-substantive changes

File: Doc/library/pickle.rst [hunk 2]
```diff
@@ -788,14 +786,16 @@ the code ::
...
-:mod:`copyreg` module is ::
+does the same but all instances of ``MyPickler`` will by default
+share the private dispatch table.  On the other hand, the code ::
 
    copyreg.pickle(SomeClass, reduce_SomeClass)
    f = io.BytesIO()
    p = pickle.Pickler(f)
 
+does the same but all instances of :class:`pickle.Pickler` will by default
+share the modified global dispatch table.
```
Comment (JelleZijlstra) [inline Doc/library/pickle.rst]: Is this change intentional? Doesn't look like a typo fix, though I agree that the old wording doesn't make much sense. However, I would word it as 
```suggestion
modifies the global dispatch table shared by all users of the :mod:`copyreg` module.
```

Comment (geryogam) [reply Doc/library/pickle.rst]: Yes this was intentional but your wording is better, thanks!

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Looks good, I'll merge in the coming days.

Draft commit message: "pickle docs: Fix typos and improve wording"

**Comment by maggyero** [comment]
Thanks for the review!
